test/SlowTests/Issues/RavenDB-18373.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using FastTests;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Operations.Indexes;
+using Raven.Client.Documents.Session;
+using Raven.Client.Exceptions;
+using Raven.Client.Exceptions.Documents;
+using Voron;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_18373 : RavenTestBase
+    {
+        public RavenDB_18373(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        class TestObj
+        {
+            public string Id { get; set; }
+            public string Prop { get; set; }
+        }
+
+        private class Simple_Map_Index : AbstractIndexCreationTask<TestObj>
+        {
+            public Simple_Map_Index()
+            {
+                Map = companies => from c in companies
+                    select new
+                    {
+                        Name = ""
+                    };
+            }
+        }
+
+        [Theory]
+        [InlineData(TransactionMode.ClusterWide)]
+        [InlineData(TransactionMode.SingleNode)]
+        public async Task Waiting_For_Index_Doesnt_Work_For_ClusterWideTransaction(TransactionMode transactionMode)
```

Comment:
Let's also add a test that is doing a delete and check that is it working as expected.

Reply:
Done